guardian/archivehunter
# Rebuild UI with Typescript & Material UI

## What does this change?
- switch to Typescript instead of Babel
- re-implement the core searching functionality, it works much better now
- deprecate react-treebears, react-multistep, react-radiogroup and others; standardise on Material-UI
- use modern CSS like grids and flexboxes wherever possible
- streamline and improve the UI where possible through greater use of tooltips and icon buttons

## How to test
The main functionality and operating logic should be unchanged, but the overall display should be neater. Once the frontend is built, navigate through and try all the usual operations

## How can we measure success?
- less deprecation warnings and other problems with outdated components
- quicker development through type-safety and css-in-javascript
- fixing of edge case irritations like the tree view misbehaving on opening and closing and overlapping searches showing irrelevant items

## Have we considered potential risks?
Key users have been informed and asked to report any issues they see promptly so we can investigate.  As much testing as possible is being carried out in Dev prior to live deployment
